nitorch/tools/preproc.py

```diff
@@ -212,37 +306,71 @@ def reset_origin(pth_in, vx=None, prefix='o', device='cpu', interpolation='linea
     """
     device = _get_device(device)
     # Reslice image data to world FOV
-    pth_in, dat, M0 = reslice2world(pth_in, vx=vx, prefix=prefix, device=device, write=False,
+    pth, dat, M0 = reslice2world(pth, vx=vx, prefix=prefix, device=device, write=False,
                                  interpolation=interpolation, bound=bound)
     # Compute new, reset, affine matrix
     dim = torch.tensor(dat.shape, dtype=torch.float64)
     vx = voxel_size(M0)
     if M0[:3, :3].det() < 0:
         vx[0] = - vx[0]
-    orig = (dim[:3] + 1) / 2
-    off = -vx * orig
-    affine = torch.tensor([[vx[0], 0, 0, off[0]],
-                           [0, vx[1], 0, off[1]],
-                           [0, 0, vx[2], off[2]],
-                           [0, 0, 0, 1]], dtype=torch.float64)
+    affine = get_origin_mat(dim, vx, dtype=torch.float64)
     # Write reset image
-    pth_out = write_img(pth_in, dat, affine, prefix=prefix)
+    pth_out = write_img(pth, dat, affine, prefix=prefix)
 
     return pth_out
 
 
-def reslice2world(pth_in, vx=None, prefix='r', device='cpu', write=True, interpolation='linear', bound='zero'):
+def reslice_dat(dat, affine, dim_out, interpolation='linear', bound='zero', extrapolate=False):
```

Comment:
(side note) We'll need some of that stuff in `nitorch.spatial` at some point

Reply:
Okay